pkg/boot/catalog.go

```diff
@@ -42,8 +42,18 @@ func GetCatalog(
 					serviceID,
 				)
 			}
-			services = append(services, svc)
-			usedServiceIDs[serviceID] = moduleName
+
+			filteredPlans := []service.Plan{}
+			for _, plan := range svc.GetPlans() {
+				if plan.GetStability() >= catalogConfig.MinStability {
+					filteredPlans = append(filteredPlans, plan)
+				}
+			}
+			if len(filteredPlans) > 0 {
+				svc.SetPlans(filteredPlans)
+				services = append(services, svc)
+				usedServiceIDs[serviceID] = moduleName
```

Comment:
Prior to this, services and plans were immutable at runtime. I prefer to keep it that way because rationalizing about immutable things is just easier...

I think something like this would work:

```
services = append(
  services,
  NewService(
    svc.GetProperties(),
    svc.GetServiceManager(),
    filteredPlans...,
  ),
)
usedServiceIDs[serviceID] = moduleName
```

Reply:
@krancour updated.